lib/components/viewers/pattern-viewer.tsx

```diff
@@ -67,6 +67,23 @@ const PatternViewer = ({
   // @ts-expect-error TODO: add type to ComponentContext
   const { ModeIcon, RouteRenderer } = useContext(ComponentContext)
 
+  const routePatternKeys = route?.patterns && Object.keys(route?.patterns)
+
+  // If the patternId does not exist in the route, course correct back to a valid pattern.
+  // (ex. the URL was /route/123/undefined)
+  if (
+    viewedRoute?.patternId &&
+    !routePatternKeys?.includes(viewedRoute?.patternId) &&
+    routePatternKeys &&
+    routePatternKeys.length > 0
+  ) {
+    // Set the patternId to the first pattern in the route (this will reload the page).
+    setViewedRoute({
+      patternId: routePatternKeys[0],
```

Comment:
One thing I had been thinking for a while is to turn these arrow buttons into links to the first pattern that normally appears when you open the pattern viewer (I think it is not necessarily the pattern at index zero). It's probably a bit more involved but would also clean up the code and the navigation. What do you think?

Reply:
Can you clarify what buttons you mean? I thought at first you meant these, but I checked and they're already links
<img width="345" alt="image" src="https://github.com/user-attachments/assets/243a0b9d-2377-45a7-b0a7-d456104e8499" />

Reply:
Okay, it must have been a while ago so I forgot!... I recall there is complicated piece of code to order the patterns but that is done u[on fetching the info for a route.